tangle-fragment-api/src/main/kotlin/tangle/fragment/arg.kt

```diff
@@ -17,8 +17,35 @@ package tangle.fragment
 
 import androidx.fragment.app.Fragment
 
-public fun <A : Any> Fragment.arg(bundleKey: String): Lazy<A> = lazy {
+/**
+ * Lazily retrieve an argument passed to this fragment. You should generally prefer to use the
+ * inline version.
+ *
+ * @param bundleKey key used to pass this argument
+ * @param clazz type class expected for this argument
+ * @see FragmentInjectFactory for instantiating fragments with arguments
+ */
+public fun <A : Any?> Fragment.arg(bundleKey: String, clazz: Class<A>): Lazy<A> = lazy {
```

Comment:
I'm not willing to introduce an overload for Java compatibility, considering that this library is tied to a Kotlin compiler plugin.